### PR TITLE
Testing nav drawer

### PR DESCRIPTION
**Pull request by sandarumk**
Few tests are added to test the navigation drawer which is newly implemented

### CODE REVIEW

**Comment by sandarumk** [comment]
@misaochan there are some changes to the gradle file which is added to enable espresso ui testing. Please check whether this does not affect any other tests.

**Comment by sandarumk** [comment]
changes related to #586

File: app/build.gradle [hunk 1]
```diff
@@ -24,7 +24,15 @@ dependencies {
 
     testCompile 'junit:junit:4.12'
     androidTestCompile "com.android.support:support-annotations:${project.supportLibVersion}"
-    androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.2'
+   // androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.2'
+
```
Comment (whym) [inline app/build.gradle]: Why would you comment it out?

File: app/src/androidTest/java/fr/free/nrw/commons/NavigationDrawerTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    @Test
+    public void testNavigationDrawer(){
+        onView(withId(R.id.drawer_layout))
+                .check(matches(isClosed(Gravity.LEFT)))
+                .perform(open());
+
+        //test the settings fragment is displayed
+        onView(withId(R.id.settings_item)).perform(click());
```
Comment (whym) [inline app/src/androidTest/java/fr/free/nrw/commons/NavigationDrawerTest.java]: This causes "Error:(40, 27) error: cannot find symbol variable settings_item", if you rebase master. After some recent changes around the drawer, this patch seems to need some revision.

**Comment by whym** [comment]
@sandarumk  Sorry for not coming back to this earlier, but it looks like in the meantime some changes affecting the drawer made the patch obsolete.

I do love the idea of having a test for it. Perhaps you might want to drop this and make a fresh try later. Of course, you are welcome to amend it to accomodate the changes, if you prefer that way. https://www.codacy.com/app/domdomegg/apps-android-commons/pullRequest?prid=661305 has some suggestion about coding style, too.

**Comment by neslihanturan** [comment]
If there won't be any revisions ,n a few days, I think we can close this PR.

**Comment by misaochan** [comment]
Agreed, closing. Thanks all!

**Pull request closed without merging**